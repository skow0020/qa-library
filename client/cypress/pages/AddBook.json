{
    "title": "#title",
    "author": "#author",
    "category": "#category",
    "language": "#language",
    "url": "#url",
    "backgroundImage": "#backgroundImage",
    "body": "#body",
    "pdfUrl": "#pdf"
}