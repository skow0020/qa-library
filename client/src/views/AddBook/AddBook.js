/* eslint jsx-a11y/anchor-is-valid: 0 */

import AlertModal, { showAlert } from "components/common/AlertModal";
import {
  Button,
  Col,
  Container,
  Form,
  FormInput,
  FormTextarea,
  Row
} from "shards-react";

import CategoriesSelection from "components/common/CategoriesSelection";
import LanguagesSelection from "components/common/LanguagesSelection";
import PageTitle from "components/common/PageTitle";
import React from "react";
import axios from "axios";

class AddBook extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      backgroundImage: '',
      author: '',
      category: 'General',
      url: '',
      title: '',
      body: '',
      pdf: '',
      language: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleCategoryChange = this.handleCategoryChange.bind(this);
    this.handleLanguageChange = this.handleLanguageChange.bind(this);
  }

  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  handleCategoryChange(e) {
    this.setState({ category: e.target.value });
  }

  handleLanguageChange(e) {
    this.setState({ language: e.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    const book = {
      backgroundImage: this.state.backgroundImage,
      author: this.state.author,
      category: this.state.category,
      url: this.state.url,
      title: this.state.title,
      body: this.state.body,
      pdf: this.state.pdf,
      language: this.state.language
    };

    axios.post("/api/books/", book)
      .then(
        response => {
          if (response.data.success) { this.showNotifier("Book added successfully"); }
          else { this.showNotifier(`Unable to add book: ${JSON.stringify(response.data.error.errors)}`); }
          this.nextPath('/books');
        },
        error => this.showNotifier(`Unable to add book: ${error}`)
      );
  }

  nextPath(path) {
    setTimeout(function () {
      this.props.history.push(path);
    }.bind(this), 2000);
  }

  showNotifier = (message) => {
    showAlert({ message });
  }

  render() {
    const { url, title, author, backgroundImage, pdf, category, body, language } = this.state;

    return (
      <Container fluid className="main-content-container px-4">
        <AlertModal />
        <Form id="add-book-form" onSubmit={this.handleSubmit}>
          <Row noGutters className="page-header py-4">
            <PageTitle sm="8" title="Add a Book" className="text-sm-left" />
            <Button id="submit" type="submit" className="btn btn-success btn-lg">
              Submit
            </Button>
          </Row>

          <label className="text-muted font-weight-bold" htmlFor="title">Book Title</label>
          <FormInput id="title" type="text" value={title} onChange={this.handleChange} required />
          <Row form>
            <Col md="4" className="form-group">
              <label className="text-muted font-weight-bold" htmlFor="author">Book Author</label>
              <FormInput id="author" type="text" value={author} onChange={this.handleChange} required />
            </Col>
            <LanguagesSelection id="language" value={language} onChange={this.handleLanguageChange} />
            <CategoriesSelection id="category" value={category} onChange={this.handleCategoryChange} required />
          </Row>
          <label className="text-muted font-weight-bold" htmlFor="url">Book URL</label>
          <FormInput id="url" type="text" value={url} onChange={this.handleChange} required />
          <label className="text-muted font-weight-bold" htmlFor="backgroundImage">Background Image URL</label>
          <FormInput id="backgroundImage" type="text" value={backgroundImage} onChange={this.handleChange} required />
          <label className="text-muted font-weight-bold" htmlFor="pdf">Book PDF URL</label>
          <FormInput id="pdf" type="text" value={pdf} onChange={this.handleChange} />
          <label className="text-muted font-weight-bold" htmlFor="body">Book Description</label>
          <FormTextarea id="body" type="textarea" style={{ height: "150px" }} value={body} onChange={this.handleChange} />
        </Form>
      </Container>
    );
  }
}

export default AddBook;
