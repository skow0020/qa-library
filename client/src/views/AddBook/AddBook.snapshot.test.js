import AddBook from './AddBook';
import React from 'react';
import renderer from 'react-test-renderer';

describe('AddBook Snapshot Tests', () => {
  test('AddBook snapshot', () => {
    const component = renderer.create(
      <AddBook></AddBook>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});