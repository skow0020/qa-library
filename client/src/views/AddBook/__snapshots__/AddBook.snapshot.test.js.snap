// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBook Snapshot Tests AddBook snapshot 1`] = `
<div
  className="main-content-container px-4 container-fluid"
>
  <div />
  <form
    className=""
    id="add-book-form"
    onSubmit={[Function]}
  >
    <div
      className="page-header py-4 no-gutters row"
    >
      <div
        className="text-sm-left text-center text-md-left mb-sm-0 col-12 col-sm-8"
      >
        <span
          className="text-uppercase page-subtitle"
        />
        <h3
          className="page-title"
        >
          Add a Book
        </h3>
      </div>
      <button
        className="btn btn-success btn-lg btn btn-primary"
        id="submit"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
    <label
      className="text-muted font-weight-bold"
      htmlFor="title"
    >
      Book Title
    </label>
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <div
      className="form-row"
    >
      <div
        className="form-group col-md-4"
      >
        <label
          className="text-muted font-weight-bold"
          htmlFor="author"
        >
          Book Author
        </label>
        <input
          className="form-control"
          id="author"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="MuiFormControl-root makeStyles-formControl-8"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="language-selection"
        >
          Languages
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby=" category"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
            id="category"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            onAnimationStart={[Function]}
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiFormControl-root makeStyles-formControl-91"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="category"
        >
          Categories
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
          required={true}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby=" language"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
            id="language"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            General
          </div>
          <input
            onAnimationStart={[Function]}
            type="hidden"
            value="General"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <label
      className="text-muted font-weight-bold"
      htmlFor="url"
    >
      Book URL
    </label>
    <input
      className="form-control"
      id="url"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="text-muted font-weight-bold"
      htmlFor="backgroundImage"
    >
      Background Image URL
    </label>
    <input
      className="form-control"
      id="backgroundImage"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="text-muted font-weight-bold"
      htmlFor="pdf"
    >
      Book PDF URL
    </label>
    <input
      className="form-control"
      id="pdf"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      className="text-muted font-weight-bold"
      htmlFor="body"
    >
      Book Description
    </label>
    <textarea
      className="form-control"
      id="body"
      onChange={[Function]}
      style={
        Object {
          "height": "150px",
        }
      }
      type="textarea"
      value=""
    />
  </form>
</div>
`;
